### PR TITLE
Refactor Forgot and Change password API

### PR DESCRIPTION
**Pull request by priyanshisharma**
### Description
Forgot Password and Change Password APIs have been updated taking Teesco backend as reference.

**Type of Change:** (Delete irrelevant options)
- Code
- Documentation

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
I've tested them using Swagger UI.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Review by anshsrtv — APPROVED** [review]
Great job writing the APIs 🌮 🎉  I would, however, suggest that the swagger docs can be updated a little bit. There are some differences in what has been documented and what is actually returned from the API, for instance, the responses from APIs do not have the `detail` key as mentioned in the docs.

File: users/views.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
+import datetime
+from datetime import timedelta
```
Comment (sahil9001) [inline users/views.py]: This isn't used anywhere, so I think it should be removed

File: users/views.py [hunk 2]
```diff
@@ -79,29 +81,91 @@ def post(self, request):
...
+
+        # Checking if the email entered is valid.
+        if serializer.is_valid():
+            valid_data = serializer.validated_data
+
+            # Check if such a user exists.
+            try:
+                user = CustomUser.objects.get(email=valid_data['email'])
+            except CustomUser.DoesNotExist:
+                return Response("Account with this email id doesn't exists. Kindly signup.", status.HTTP_404_NOT_FOUND)
```
Comment (sahil9001) [inline users/views.py]:
```suggestion
                return Response(responses.change_password_404, status.HTTP_404_NOT_FOUND)
```

File: users/views.py [hunk 3]
```diff
@@ -79,29 +81,91 @@ def post(self, request):
...
+            else:
+                otp = str(randint(1000, 9999))
+                user.otp = otp
+                user.save()
+                send_email_otp(recipient_list=[user.email], otp=otp)
+
+                return Response("An otp has been sent to you to reset your password", status.HTTP_200_OK)
```
Comment (sahil9001) [inline users/views.py]:
```suggestion
                return Response(responses.forgot_password_200, status.HTTP_200_OK)
```

I think responses should be used, I don't know if we just return the message directly.

File: users/views.py [hunk 4]
```diff
@@ -79,29 +81,91 @@ def post(self, request):
...
+        serializer = ChangePasswordSerializer(data=request.data)
+
+        if serializer.is_valid():
+            valid_data = serializer.validated_data
+
+            # Check if such a user exists.
+            try:
+                user = CustomUser.objects.get(email=valid_data['email'])
+            except CustomUser.DoesNotExist:
+                return Response("Account with this email id doesn't exists. Kindly signup.", status.HTTP_404_NOT_FOUND)
```
Comment (sahil9001) [inline users/views.py]:
```suggestion
                return Response(responses.change_password_404, status.HTTP_404_NOT_FOUND)
```
I think responses should be used, I don't know if we just return the message directly.

File: users/views.py [hunk 5]
```diff
@@ -79,29 +81,91 @@ def post(self, request):
...
+            
+            if valid_data['otp'] == user.otp:
+                user.set_password(valid_data['password'])
+                user.save()
+                return Response('Password changed successfully', status.HTTP_200_OK)
```
Comment (sahil9001) [inline users/views.py]:
```suggestion
                return Response(responses.change_password_200, status.HTTP_200_OK)
```
I think responses should be used, I don't know if we just return the message directly.

File: users/views.py [hunk 6]
```diff
@@ -79,29 +81,91 @@ def post(self, request):
...
+            else:
+                return Response('Invalid otp', status.HTTP_401_UNAUTHORIZED)
```
Comment (sahil9001) [inline users/views.py]:
```suggestion
                return Response(responses.change_password_401, status.HTTP_401_UNAUTHORIZED)
```

**Review by sahil9001 — CHANGES_REQUESTED** [review]

**Review by sahil9001 — APPROVED** [review]

**Comment by sahil9001** [comment]
It's looking great!

**Pull request merged**